### PR TITLE
feat: Swarm.RelayService.InfiniteLimits

### PR DESCRIPTION
**Pull request by lidel**
This PR needs https://github.com/ipfs/kubo/pull/9656 to land first.

## Context 

Allowing users to switch from additional v1 relays to built-in v2 in Kubo.

The idea here is to expose https://github.com/libp2p/go-libp2p/pull/2125 as an experimental, opt-in configuration option, 
for running an unlimited relay in controlled environments (when relay and clients are managed by the same entity). 

We want to fix/update tutorial from https://github.com/ipfs/ipfs-docs/issues/1459. If we can remove the need for asking people to run a separate [libp2p-relay-daemon v1](https://github.com/libp2p/go-libp2p-relay-daemon)), and only run Kubo, that it is a big  win and reduction in complexity and DX/UX win.

## TODO

- [ ] rebase after https://github.com/ipfs/kubo/pull/9656 lands
- [ ] confirm bitswap works and this option helps with https://github.com/ipfs/ipfs-docs/issues/1459 
- [ ] add tests (tbd if sharness, or https://github.com/ipfs/interop/, need to sync with Alex about relays and Helia)

cc @2color  @thediscordian for visibility

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -67,7 +67,7 @@ require (
 	github.com/jbenet/go-temp-err-catcher v0.1.0
 	github.com/jbenet/goprocess v0.1.4
 	github.com/libp2p/go-doh-resolver v0.4.0
-	github.com/libp2p/go-libp2p v0.25.1
+	github.com/libp2p/go-libp2p v0.25.2-0.20230222220431-c03190e1ca4f
```
Comment (lidel) [inline go.mod:70]: :information_source:  switch to release after https://github.com/ipfs/kubo/pull/9656 lands.

File: config/swarm.go [hunk 2]
```diff
@@ -93,6 +93,14 @@ type RelayService struct {
 	MaxReservationsPerIP *OptionalInteger `json:",omitempty"`
 	// MaxReservationsPerASN is the maximum number of reservations origination from the same ASN.
 	MaxReservationsPerASN *OptionalInteger `json:",omitempty"`
+
+	// InfiniteLimits effectively removes all limits. Enabling this option also
+	// doesn't apply any filter for which protocols can be used, allowing for
+	// expensive things like bitswap over a relayed connection.
```
Comment (lidel) [inline config/swarm.go:99]: :thought_balloon: need to confirm bitswap works

**Comment by hsanjuan** [comment]
@lidel we still want this?

**Comment by lidel** [comment]
Unlikely, let's close this. 

A lot changed in ecosystem and go-libp2p. If anyone feels "unlimited relay for bitswap" is useful, they can fill an issue and we can discuss more modern approach.

----

Digression / historical context:

I think the problem I aimed to solve here was the fact that Relay v1 allowed Bitswap, and Relay v2 did not and we did not have hole punching and [DCUtR](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md) at the time so people did not have 1:1 replacement + we broke example from docs.

The idea was that if someone wanted to run their own static relay v2 infrastructure and cover the cost of relaying Bitswap over their servers, they should have that option.

But in practice, existing users kept using Relay v1 by running https://github.com/libp2p/go-libp2p-relay-daemon and never asked for this feature to land, and now that we have [DCUtR](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md) the interest is even smaller.

[DCUtR](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md) is not a silver bullet, and there will be cases where % of users is not able to do bitswap, but those days we don't want Kubo to support such edge case – a custom implementation built with go-libp2p or boxo should be used.

**Pull request closed without merging**